bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -230,14 +255,60 @@ pub mod pallet {
 				.saturating_add(T::BackendWeightInfo::transact_add_tip())
 		)]
 		pub fn add_tip(origin: OriginFor<T>, message_id: MessageId, asset: Asset) -> DispatchResult {
-			ensure_signed(origin)?;
-			// Check tip is DOT
-
-			// Convert to ether
+			let who = ensure_signed(origin)?;
+
+			let ether_location = T::EthereumLocation::get();
+			let (tip_asset_location, tip_amount) = match asset {
+				Asset {
+					id: AssetId(ref loc),
+					fun: Fungibility::Fungible(amount),
+				} => (loc, amount),
+				_ => return Err(Error::<T>::UnsupportedAsset.into()),
+			};
+
+			// Swap tip asset to ether
+			let swap_path = vec![(*tip_asset_location).clone(), ether_location.clone()];
+			let who_location = Self::account_to_location(who.clone())?;
+
+			let balance_before = T::ForeignToken::balance(ether_location.clone(), &who);
+			T::Swap::swap_exact_tokens_for_tokens(
+				who.clone(),
+				swap_path,
+				tip_amount,
+				None,
+				who.clone(),
+				true,
+			)?;
+			let balance_after = T::ForeignToken::balance(ether_location.clone(), &who);
+			let ether_gained = balance_after.saturating_sub(balance_before);
+
+			// Burn the ether
+			let ether_asset = Asset::from((ether_location.clone(), ether_gained));
+			let xcm_context = XcmContext {
+				origin: None,
+				message_id: Default::default(),
+				topic: None,
+			};
+			T::AssetTransactor::can_check_out(&who_location, &ether_asset, &xcm_context).map_err(|_| Error::<T>::WithdrawError)?;
+			T::AssetTransactor::check_out(&who_location, &ether_asset, &xcm_context);
+			T::AssetTransactor::withdraw_asset(&ether_asset, &who_location, None).map_err(|_| Error::<T>::WithdrawError)?;
```

Comment:
There is a helper function `snowbridge_core::burn_for_teleport` which can do this.